wlm/src/main/java/ai/djl/serving/wlm/util/ModelServerListener.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.serving.wlm.util;
+
+import ai.djl.Device;
+import ai.djl.serving.wlm.Adapter;
+import ai.djl.serving.wlm.ModelInfo;
+
+import java.nio.file.Path;
+
+/** An interface that represent a model server event listener. */
+public interface ModelServerListener {
```

Comment:
Do we want to create a `ModelServerListenerAdapter` to provide empty definitions (like TrainingListenerAdapter)?

Reply:
Added `ModelServerListenerAdapter`